lib/spack/spack/solver/asp.py

```diff
@@ -518,9 +554,37 @@ def solve(
 
         # set up the problem -- this generates facts and rules
         self.assumptions = []
+        self.multi_root = multi_root
         with self.control.backend() as backend:
             self.backend = backend
             solver_setup.setup(self, specs, tests=tests, reuse=reuse)
+
+            strings = []
```

Comment:
Fixed in [22f2616](https://github.com/spack/spack/pull/27964/commits/22f2616f4ccee3b54aae6cdf5dc4a8c030a7e153)[](https://github.com/becker33)
@[becker33](https://github.com/becker33) becker33 [force-pushed](https://github.com/spack/spack/compare/c5af6698965475cb483d5ba109d0be4884418aa5..22f2616f4ccee3b54aae6cdf5dc4a8c030a7e153) the features/concretize-together-conflicts branch from [c5af669 ](https://github.com/spack/spack/commit/c5af6698965475cb483d5ba109d0be4884418aa5)to [22f2616 ](https://github.com/spack/spack/commit/22f2616f4ccee3b54aae6cdf5dc4a8c030a7e153)[now](https://github.com/spack/spack/pull/27964#event-6023437721)
Merge state
Add more commits by pushing to the [features/concretize-together-conflicts](https://github.com/spack/spack/tree/features/concretize-together-conflicts) branch on [spack/spack](https://github.com/spack/spack)[](https://github.com/apps/github-actions).

This branch has not been deployed
No deployments
This pull request is still a work in progress
Draft pull requests cannot be merged.
Some checks haven’t completed yet
23 queued, 14 expected, and 1 pending checks
@github-actions
Bootstrapping / fedora-clingo-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842080?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
linux tests / validate (pull_request) Queued — Waiting to run this check...
Required
[Details](https://github.com/spack/spack/runs/5102842066?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842744?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
linux tests / style (pull_request) Queued — Waiting to run this check...
Required
[Details](https://github.com/spack/spack/runs/5102842132?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries-and-patchelf (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842713?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
linux tests / changes (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102841889?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / opensuse-clingo-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842481?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102841956?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-binaries (3.5) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842129?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-binaries (3.6) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842191?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-binaries (3.7) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842229?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-binaries (3.8) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842280?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-clingo-binaries (3.9) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842317?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (2.7) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842519?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (3.5) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842552?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (3.6) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842579?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (3.7) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842601?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (3.8) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842631?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-clingo-binaries (3.9) (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842668?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-gnupg-binaries (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842871?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / ubuntu-gnupg-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842774?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-gnupg-binaries (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842363?check_suite_focus=true)[](https://github.com/apps/github-actions)
@github-actions
Bootstrapping / macos-gnupg-sources (pull_request) Queued — Waiting to run this check...
[Details](https://github.com/spack/spack/runs/5102842437?check_suite_focus=true)[](https://readthedocs.org/)
build (3.8) Expected — Waiting for status to be reported
Required
ci/gitlab-ci Expected — Waiting for status to be reported
Required
clingo-cffi Expected — Waiting for status to be reported
Required
package-audits Expected — Waiting for status to be reported
Required
shell Expected — Waiting for status to be reported
Required
unittests (2.7, clingo) Expected — Waiting for status to be reported
Required
unittests (2.7, original) Expected — Waiting for status to be reported
Required
unittests (3.5, clingo) Expected — Waiting for status to be reported
Required
unittests (3.6, clingo) Expected — Waiting for status to be reported
Required
unittests (3.6, original) Expected — Waiting for status to be reported
Required
unittests (3.7, clingo) Expected — Waiting for status to be reported
Required
unittests (3.8, clingo) Expected — Waiting for status to be reported
Required
unittests (3.9, clingo) Expected — Waiting for status to be reported
Required
unittests (3.9, original) Expected — Waiting for status to be reported
Required

docs/readthedocs.org:spack Pending — Read the Docs build is in progress!
Required
[Details](https://readthedocs.org/projects/spack/builds/16015744/)
 You can also [open this in GitHub Desktop](https://desktop.github.com/)[](https://github.com/becker33)[](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) or view .
@becker33

 
 
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/spack/spack/blob/e97bffc3e833b6fe7ad561c85fbe177ca6647e95/.github/CONTRIBUTING.md) and [code of conduct](https://github.com/spack/spack/blob/e97bffc3e833b6fe7ad561c85fbe177ca6647e95/.github/CODE_OF_CONDUCT.md).
 ProTip! Add [.patch](https://github.com/spack/spack/pull/27964.patch) or [.diff](https://github.com/spack/spack/pull/27964.diff)[](https://github.com/alalazo) to the end of URLs for Git’s plaintext views.
Reviewers
@[alalazo](https://github.com/alalazo)[](https://github.com/spack/spack/pull/27964/files/296861d60ee9b3bed3b69e2f34537ede2e9c6158)[](https://github.com/tgamblin)
alalazo
@[tgamblin](https://github.com/tgamblin)[](https://github.com/tgamblin)
[tgamblin](https://github.com/tgamblin)[](https://github.com/alalazo)
At least 1 approving review is required to merge this pull request.

Assignees
@tgamblin
tgamblin
@[alalazo](https://github.com/alalazo)
alalazo
Labels
[commands](https://github.com/spack/spack/labels/commands)
[dependencies](https://github.com/spack/spack/labels/dependencies)
[environments](https://github.com/spack/spack/labels/environments)
[new-variant](https://github.com/spack/spack/labels/new-variant)
[tests](https://github.com/spack/spack/labels/tests)
Projects
https://github.com/spack/spack/projects/48#card-74706979
  
In progress 
https://github.com/spack/spack/projects/45#card-74706978[](https://github.com/becker33)[](https://github.com/alalazo)[](https://github.com/tgamblin)[](https://github.com/)
  
In progress 
Milestone
No milestone
Linked issues
Successfully merging this pull request may close these issues.

None yet

Notifications
Customize
You’re receiving notifications because you’re watching this repository.
3 participants
@becker33
@alalazo
@tgamblin